src/core_init_atmosphere/mpas_init_atm_thompson_aerosols.F

```diff
@@ -0,0 +1,889 @@
+    sorted_arr(2,1:nGocartLevels) = 0._RKIND
+    do k = 1, nGocartLevels
+       kk = nGocartLevels + 1 -k
+       sorted_arr(1,kk) = pwif_int(k,iCell)
+       sorted_arr(2,kk) = nwfa_int(k,iCell)
+    enddo
+    do k = nVertLevels, 1, -1
+       target_p = pressure(k,iCell)
+       nwfa(k,iCell) = vertical_interp(target_p,nGocartLevels-1, &
+                                sorted_arr(:,1:nGocartLevels-1),order=1,extrap=0)
+       if(target_p >= pwif_int(1,iCell) .and. k < nVertLevels) nwfa(k,iCell) = nwfa(k+1,iCell)
+    enddo
+ enddo
+
+!--- deallocation of local arrays:
+ if(allocated(dummy1)    ) deallocate(dummy1    )
+ if(allocated(dummy2)    ) deallocate(dummy2    )
+ if(allocated(nifa_int)  ) deallocate(nifa_int  )
+ if(allocated(nwfa_int)  ) deallocate(nwfa_int  )
+ if(allocated(pwif_int)  ) deallocate(pwif_int  )
+ if(allocated(sorted_arr)) deallocate(sorted_arr)
+ 
+!call mpas_log_write('--- end subroutine init_vinterp_gocart:')
+
+ end subroutine init_vinterp_gocart
+
+!=================================================================================================================
+ subroutine init_hinterp_gocart(block,mesh)
+!=================================================================================================================
+
+!inout arguments:
+ type(block_type),intent(inout),target:: block 
+ type (mpas_pool_type),intent(inout)  :: mesh
+
+!local variables:
+ type(dm_info),pointer:: dminfo
+ type(met_data) :: field !real*4 meteorological data.
+ type(proj_info):: proj
+
+ character(len=StrKIND):: filename_gocart
+ logical:: have_landmask
+
+ integer,pointer:: nCells
+ integer:: i,j
+ integer:: iCell,istatus,k,masked,nmonths,nInterpPoints
+ integer,dimension(5):: interp_list
+ integer,dimension(:),pointer:: landmask
+ integer,dimension(:),pointer:: mask_array
+
+ real(kind=RKIND):: fillval,maskval,msgval
+ real(kind=RKIND):: lat,lon,x,y
+ real(kind=RKIND),dimension(:),pointer    :: latCell,lonCell
+ real(kind=RKIND),dimension(:),pointer    :: latPoints,lonPoints
+ real(kind=RKIND),dimension(:,:,:),pointer:: nifa_clim,nwfa_clim,pwif_clim
+ real(kind=RKIND),dimension(:,:,:),pointer:: destField3d
+
+ real(kind=RKIND),dimension(:,:),allocatable:: maskslab,rslab
+
+!-----------------------------------------------------------------------------------------------------------------
+!call mpas_log_write('--- enter subroutine init_hinterp_gocart:')
+
+ dminfo => block%domain%dminfo
+
+ filename_gocart = "QNWFA_QNIFA_SIGMA_MONTHLY.dat"
+
+ call mpas_pool_get_dimension(mesh,'nCells',nCells)
+
+ call mpas_pool_get_array(mesh,'landmask',landmask)
+ call mpas_pool_get_array(mesh,'latCell' ,latCell )
+ call mpas_pool_get_array(mesh,'lonCell' ,lonCell )
+
+ call mpas_pool_get_array(mesh,'nifa_gocart_clim',nifa_clim)
+ call mpas_pool_get_array(mesh,'nwfa_gocart_clim',nwfa_clim)
+ call mpas_pool_get_array(mesh,'pwif_gocart_clim',pwif_clim)
+
+!open intermediate file:
+ istatus = 0
+ call read_gocart_init(trim(filename_gocart),istatus)
```

Comment:
I do not recall  what I was exactly thinking when I added subroutine read_gocart_init. Maybe 1) the filename does not have a time stamp and therefore "simpler" to open? and 2) wanting to keep everything in one module to keep the thompson/gocart sourcecode compact.

Reply:
I think it would be better to use the `read_met_init` routine. You can see in that routine that if the `source_is_constant` arugment is `.true.`, we use the filename as specified rather than adding the date suffix:
https://github.com/MPAS-Dev/MPAS-Model/blob/v8.1.0/src/core_init_atmosphere/mpas_init_atm_read_met.F#L72-L76

Reply:
replaced calls to subroutine read_gocart_init with calls to subroutine read_met_init, as suggested.